p2p/host/basic/basic_host.go

```diff
@@ -41,13 +43,16 @@ type BasicHost struct {
 	natmgr  *natManager
 
 	proc goprocess.Process
+
+	bwc metrics.Reporter
 }
 
 // New constructs and sets up a new *BasicHost with given Network
-func New(net inet.Network, opts ...Option) *BasicHost {
+func New(net inet.Network, bwc metrics.Reporter, opts ...Option) *BasicHost {
```

Comment:
the `metrics.Reporter` should be either an option (if it _must_ be in the constructor). or a setter:

```go
```